Adding PMT Analysis

pietro14 opened the pull request:
Now, camera and PMT analysis can be performed together or separately.

You can use the flags `"camera_mode"` and `"pmt_mode"` to choose which analysis to perform.

If both flags are set to `1`, the final output file will contain 3 trees:

1. **Events**: A tree containing reconstructed quantities for the camera.
2. **PMT_Events**: A tree containing reconstructed quantities for the PMTs.
3. **PMT_Avg_Events**: A tree containing the average PMT waveforms of 4 channels.

In the future, the following improvements will be needed:

- Integration of Bayesian fitting to associate each waveform with a single cluster.
- Merging the 3 trees into one single tree where each event contains information about a single cluster and the associated waveform (note that currently, an event is an image).

For more information about the PMT analysis, please contact David.

*(The code has been tested and runs smoothly.)*

pietro14 committed:
Fix reading n. events in LNF runlog file

pietro14 committed:
Merge branch 'CYGNUS-RD:winter23' into winter23

pietro14 committed:
Update PMT analysis

pietro14 committed:
Bug fix: bank key in waveform reading

pietro14 committed:
Uncomment saving gitHash

pietro14 committed:
Add try/except in conversion_env_variables()

pietro14 committed:
Merge branch 'pmt_reco_w23' of https://github.com/pietro14/reconstruction into pmt_reco_w23

pietro14 committed:
Reverse utilities.py

pietro14 committed:
Merge branch 'CYGNUS-RD:winter23' into winter23

pietro14 committed:
Merge branch 'winter23' into pmt_reco_w23

pietro14 committed:
Merge pull request #1 from pietro14/pmt_reco_w23

Pmt reco w23

pietro14 committed:
Update configFile_LNGS.txt

pietro14 committed:
Add PMT time variables + remove useless files

pietro14 committed:
Delete useless file

igorabritta commented on utilities.py:
```diff
@@ -332,49 +332,46 @@ def peak_memory_usage(self):
-            
-        return dslow
     
+    def conversion_env_variables(self, dslow, odb, i=0, j=0):
```
If it doesn't have this value for an image, the variable will be saved as "null", "0" or something else?
I'm asking because we usually use the average of these values per run to show at the Reco History Dashboard.

igorabritta commented on configFile_LNGS.txt:
```diff
@@ -19,7 +19,7 @@
 'flag_stats'            : 0,
 
-'camera_mode'            : 1,
+'camera_mode'            : 0,
```
Shouldn't 'camera_mode' stays On?

pietro14 replied:
Yes, I'll set it to 1. It was set to 0 because I was running some tests running only the pmt reco, without the camera reco.

igorabritta commented on reconstruction.py:
```diff
@@ -626,10 +629,17 @@ def reconstruct(self,evrange=(-1,-1,-1)):
                                 indx = trg * nChannels_f + ch
                                 waveform_info = { 'run' : run, 'event': event, 'channel' : ch, 'trigger' : trg , 'GE' : insideGE, 'sampling' : "fast", 'TTT' : (TTTs_f[trg]*8.5/1000./1000.)}
 
+                                t0 = time.perf_counter()
```
It could be a good idea to use different variable names for the t0 and t1.
And we could also think about cleaning the variable after each to be sure to not save old or memory values by mistake.

pietro14 committed:
Set 'camera_mode' to 1

pietro14 committed:
environment_variables set to True

pietro14 committed:
Fix event number for PMTs

This commit corrects a discrepancy in the event number between the waveforms and images.

pietro14 committed:
Undo change on setPedestlRun

Undo change on setPedestlRun function

davidjgmarques committed:
testing push

davidjgmarques committed:
Waveform script cleaned

davidjgmarques committed:
treeVars script cleaned

davidjgmarques committed:
treeVars script better clean

davidjgmarques committed:
Main reco cleaned

davidjgmarques committed:
Fixed run number issue and cleaned tree fill

pietro14 committed:
Merge pull request #2 from davidjgmarques/winter23

PMT Reco update

pietro14 committed:
Removing some comments

pietro14 committed:
Update reconstruction.py

pietro14 committed:
Update if statement on fillEnvVariables

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Update filling of Env. Variables

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes

pietro14 committed:
Don't compute pedestal when pmt mode

pietro14 committed:
Minor changes

pietro14 committed:
Minor changes and comments

pietro14 commented:
Closing the old pull request. I am going to open a new one.
